File: stack.js
// stack_menu.js
const readline = require("readline");

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let stack = [];

function showMenu() {
  //Menu Is Here
    console.log("1. Push");
    console.log("2. Pop");
    console.log("3. Peek");
    console.log("4. Display");
    console.log("5. Exit");

    rl.question("Enter your choice: ", handleChoice);
}

function handleChoice(choice) {
    switch (choice) {
        case "1": 
            rl.question("Enter element to push: ", (element) => {
                stack.push(element);
                console.log(element + " pushed into stack");
                showMenu();
            });
            break;

        case "2": 
            if (stack.length === 0) {
                console.log("Stack Underflow! Stack is empty");
            } else {
                let removed = stack.pop();
                console.log(removed + " popped from stack");
            }
            showMenu();
            break;

        case "3": // PEEK
            if (stack.length === 0) {
                console.log("Stack is empty");
            } else {
                console.log("Top element is:", stack[stack.length - 1]);
            }
            showMenu();
            break;

        case "4": // DISPLAY
            if (stack.length === 0) {
                console.log("Stack is empty");
            } else {
                console.log("Stack elements:", stack.join(" "));
            }
            showMenu();
            break;

        case "5": // EXIT
            console.log("Exiting program...");
            rl.close();
            break;

        default:
            console.log("Invalid choice! Try again");
            showMenu();
    }
}

// start program
showMenu();

File: Stack.c
#define max 5
#include <stdio.h>

int push(int top, int stack[])
{

    int x;
    if (top == max - 1)
    {
        printf("Stack is full , Cannot Push ");
    }
    else
    {
        printf("Enter value : ");
        scanf("%d", &x);
        top++;
        stack[top] = x;
    }
    return (top);
}

int pop(int top, int stack[])
{
    int x;
    if (top == -1)
    {
        printf("Empty hai yrrr..!!");
    }
    else
    {
        x = stack[top];
        printf("Value Poped : %d\n", x);
        top--;
    }
    return (top);
}

void display(int top, int stack[])
{
    if (top == -1)
    {
        printf("Stack is Underflow , Cannot Display !! ");
    }
    else
    {
        while (top >= 0)
        {
            printf("%d\n", stack[top]);
            top--;
        }
    }
}

main()
{
    int top = -1, ch, stack[max];

    do
    {
        printf("\n1. Push\n2. Pop \n3. Display\nEnter your choice : ");
        scanf("%d", &ch);
        switch (ch)
        {
        case 1:
            top = push(top, stack);
            break;
        case 2:
            top = pop(top, stack);
            break;
        case 3:
            display(top, stack);
            break;
        }
    } while (ch <= 3);
}
